Cross-compilation fails with "target may not be installed", missing std crate for #[no_std] project

I'm not sure if this is an issue with naersk, or [rust-overlay](https://github.com/oxalica/rust-overlay) which I am also using, so i've also reported an issue there: https://github.com/oxalica/rust-overlay/issues/12.

I've got the following:

<details>
  <summary>flake.nix</summary>

  ```nix
  {
    description = "A very basic flake";
  
    inputs = {
      nixpkgs.url = "nixpkgs/nixos-20.09";
  
      rust-overlay = {
        url = "github:oxalica/rust-overlay";
        inputs.nixpkgs.follows = "nixpkgs";
      };
  
      naersk = {
        url = "github:nmattia/naersk";
        inputs.nixpkgs.follows = "nixpkgs";
      };
    };
  
    outputs = { self, nixpkgs, rust-overlay, naersk }:
      let
        systems = [
          "aarch64-linux"
          "x86_64-linux"
        ];
  
        eachSystem = systems: f: builtins.foldl' (attrs: system: attrs // { ${system} = f system; }) { } systems;
      in
      {
        packages = eachSystem systems
          (system:
            let
              pkgs = import nixpkgs { inherit system; overlays = [ rust-overlay.overlay ]; };
              toolchain = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain;
            in
            {
              fw = (naersk.lib.${system}.override {
                cargo = toolchain;
                rustc = toolchain;
              }).buildPackage {
                src = ./sbs-fw;
              };
            }
          );
      };
  }
  ```
</details>

<details>
  <summary>rust-toolchain</summary>

  ```toml
  [toolchain]
  channel = "nightly-2021-01-27"
  components = [ "clippy-preview", "miri-preview", "rls-preview", "rust-analysis", "rust-analyzer-preview", "rustfmt", "llvm-tools-preview", "rust-src", "rust-std" ]
  targets = [ "thumbv7m-none-eabi", "x86_64-pc-windows-gnu", "x86_64-unknown-linux-gnu" ]
  ```
</details>

`cargo build` in the sbs-fw directory works just fine, however, when doing a `nix build .#fw` the following error occurs:
```
sbs-fw-deps>    Compiling stm32f1xx-hal v0.6.1
sbs-fw-deps> error: could not compile `sbs-fw`
sbs-fw-deps> To learn more, run the command again with --verbose.
sbs-fw-deps> warning: build failed, waiting for other jobs to finish...
sbs-fw-deps> error: build failed
sbs-fw-deps> error[E0463]: can't find crate for `std`
sbs-fw-deps>   |
sbs-fw-deps>   = note: the `thumbv7m-none-eabi` target may not be installed
sbs-fw-deps> error: aborting due to previous error
sbs-fw-deps> For more information about this error, try `rustc --explain E0463`.
sbs-fw-deps> [naersk] cargo returned with exit code 101, exiting
error: --- Error --- nix-daemon
builder for '/nix/store/azanh3lgylx1gpcdbbdsqzkjy83hp0kp-sbs-fw-deps-0.1.0.drv' failed with exit code 101; last 10 log lines:
    |
    = note: the `thumbv7m-none-eabi` target may not be installed
  
  
  error: aborting due to previous error
  
  
  For more information about this error, try `rustc --explain E0463`.
  
  [naersk] cargo returned with exit code 101, exiting
```

--- a/lib.nix
+++ b/lib.nix
@@ -177,7 +177,8 @@ rec
                 # make sure cargo is happy
                 pushd $out/$member > /dev/null
                 mkdir -p src
-                touch src/lib.rs
+                # Avoid accidentally pulling `std` for no-std crates.
+                echo '#![no_std]' >src/lib.rs
                 # pretend there's a `build.rs`, otherwise cargo doesn't build
                 # the `[build-dependencies]`. Custom locations of build scripts
                 # aren't an issue because we strip the `build` field in

